packages/repo-metadata-lint/src/validate.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {URL} from 'url';
+
+export interface ValidationResult {
+  status: 'success' | 'error';
+  errors: Array<string>;
+}
+
+const LIBRARY_TYPES = [
+  'GAPIC_AUTO',
+  'GAPIC_MANUAL',
+  'AGENT',
+  'MISC',
+  'AUTH',
+  'various',
+  'REST',
+];
+// Types of libraries that should include api_shortname in .repo-metadata.json.
+const API_LIBRARY_TYPES = ['GAPIC_AUTO', 'GAPIC_MANUAL', 'AGENT'];
```

Comment:
GAPIC_COMBO (e.g. Cloud Speech in Python or Node.js) 
It may not hold universally, and so could be complicated to validate, but 'INTEGRATION' (e.g. Spanner for Django, Java  Spring libs) would likely benefit from including the api_shortname.

Reply:
added both.